pkg/rules/engine/engine_test.go

```diff
@@ -512,3 +514,24 @@ func TestGetSelectedEvents(t *testing.T) {
 	}
 	assert.ElementsMatch(t, expected, se)
 }
+func TestLoadSignature(t *testing.T) {
+	var sigs []detect.Signature
+	input := make(chan protocol.Event)
+	source := EventSources{
+		Tracee: input,
+	}
+	output := make(chan detect.Finding)
+	engine, err := NewEngine(sigs, source, output, os.Stdout, Config{})
+	assert.NoError(t, err)
+
+	sig := regoFakeSignature{}
+	metadata, err := sig.GetMetadata()
+	assert.NoError(t, err)
+
+	id, err := engine.LoadSignature(&sig)
+	assert.NoError(t, err)
+	assert.Equal(t, metadata.ID, id)
+
+	//check that signature stats were correctly incremented
+	assert.Equal(t, 1, int(engine.Stats().Signatures.Read()))
```

Comment:
Are you confident with this test case? Maybe we could check that it works as expected for more than one signature as well?